### ISSUE TITLE
class org.eclipse.core.runtime.Plugin signer information does not match signer information

### ISSUE THREAD

**Issue by anon767** [Fraunhofer-AISEC/cpg#472]
Hi,
Lately, I receive Java SecurityExceptions when working with C/C++ Code.  It seems like some signatures from the org.eclipse.core.runtime do not match?
Strangely, it started to happen suddenly without making any changes to my build.gradle, or to any other part of my code.

Here is my build.gradle.kts. Any help would be appreciated:

```
import org.jetbrains.kotlin.gradle.tasks.KotlinCompile

plugins {
    // built-in
    java
    application
    kotlin("jvm") version "1.5.10"
}
application {
    mainClassName = "com.thecout.cpg.ApplicationKt"
}
group = "com.thecout.cpg"

repositories {
    mavenCentral()
    ivy {
        setUrl("https://download.eclipse.org/tools/cdt/releases/10.2/cdt-10.2.0/plugins")
        metadataSources {
            artifact()
        }
        patternLayout {
            artifact("/[organisation].[module]_[revision].[ext]")
        }
    }
}
java {
    sourceCompatibility = JavaVersion.VERSION_11
    targetCompatibility = JavaVersion.VERSION_11
}

val versions = mapOf(
    "junit5" to "5.6.0",
    "cpg" to "3.5.1"
)
dependencies {
    testImplementation("org.junit.jupiter", "junit-jupiter-api", versions["junit5"])
    testImplementation("org.junit.jupiter", "junit-jupiter-params", versions["junit5"])
    testRuntimeOnly("org.junit.jupiter", "junit-jupiter-engine", versions["junit5"])
    api("info.picocli:picocli:4.1.4")
    annotationProcessor("info.picocli:picocli-codegen:4.1.4")
    api("de.fraunhofer.aisec", "cpg", versions["cpg"])
}
val compileKotlin: KotlinCompile by tasks
compileKotlin.kotlinOptions {
    jvmTarget = "11"
}
val compileTestKotlin: KotlinCompile by tasks
compileTestKotlin.kotlinOptions {
    jvmTarget = "11"
}

```

Thanks in advance.

P.S.
the stacktrace of the said error:

```
Caused by: java.lang.ExceptionInInitializerError
	at org.eclipse.cdt.core.dom.parser.AbstractCLikeLanguage.createScanner(AbstractCLikeLanguage.java:279)
	at org.eclipse.cdt.core.dom.parser.AbstractCLikeLanguage.getASTTranslationUnit(AbstractCLikeLanguage.java:145)
	at de.fraunhofer.aisec.cpg.frontends.cpp.CXXLanguageFrontend.parse(CXXLanguageFrontend.java:279)
	at de.fraunhofer.aisec.cpg.TranslationManager.runFrontends(TranslationManager.java:241)
	at de.fraunhofer.aisec.cpg.TranslationManager.lambda$analyze$0(TranslationManager.java:98)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1771)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1763)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1016)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1665)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1598)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: java.lang.SecurityException: class "org.eclipse.core.runtime.Plugin"'s signer information does not match signer information of other classes in the same package
	at java.base/java.lang.ClassLoader.checkCerts(ClassLoader.java:1151)
	at java.base/java.lang.ClassLoader.preDefineClass(ClassLoader.java:906)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1015)
	at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:151)
	at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:821)
	at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:719)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:642)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:600)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1017)
	at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:151)
	at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:821)
	at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:719)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:642)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:600)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	at org.eclipse.cdt.internal.core.parser.scanner.CPreprocessor.getCDTVersion(CPreprocessor.java:139)
	at org.eclipse.cdt.internal.core.parser.scanner.CPreprocessor.<clinit>(CPreprocessor.java:105)
```

**Comment by oxisto** [comment]
We run into the same issue here. It seems that somehow eclipse meddled with already published artefacts. We are still in the process of preparing the 4.0 release. We have the first beta available as a pre-release and we are working on the second one. You can fetch them via JitPack.

**Comment by oxisto** [comment]
Small update: If you want to hop on the 4.0 train, you can use our recent, already pretty stable 4.0.0-beta.2 pre-release: https://github.com/Fraunhofer-AISEC/cpg/releases/tag/v4.0.0-beta.2. The major API change is that language frontends can now be configured individually and one must call `.defaultLanguages` when configuring the translation. Apart from that, the API should be pretty stable.

**Comment by oxisto** [comment]
I have just released 3.5.2 with a hot-fix for this, but switching to the 4.0.0-beta.2 might be the better option. It will take some time until it is propagated in Maven central.
